test/ford.hpp

```diff
@@ -53,7 +53,7 @@ std::future<typename std::result_of<Function()>::type> detach_async(Function&& f
     std::packaged_task<result_type()> task(std::forward<Function>(f));
     auto fut = task.get_future();
     std::thread(std::move(task)).detach();
-    return std::move(fut);
+    return fut;
```

Comment:
same as above two

Reply:
@JehandadKhan IIRC since C++11 std::move prevents copy elision. std::move is useless because when copy elision is not applicable, the compiler should try move constructor. Ditto two other cases.